posts/views.py

```diff
@@ -85,3 +88,76 @@ def put(self, request, author_id, post_id):
             return Response(status=status.HTTP_204_NO_CONTENT)
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class PostList(ListCreateAPIView):
+    serializer_class = PostSerializer
+    pagination_class = PostsPagination
+
+    # used by the ListCreateAPIView super class 
+    def get_queryset(self):
+        return self.posts
+
+    """
+    Get recent posts of author (paginated)
+    """
+    def get(self, request, *args, **kwargs):
+        try:
+            self.posts = Post.objects.filter(
+                author_id=kwargs.get("author_id")
+            ).order_by('-published')
+        except (KeyError, Post.DoesNotExist):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+ 
+        response = super().list(request, *args, **kwargs)
+        return response
+    
+    """
+    Create a Post with generated post id
+    """
+    def post(self, request, author_id):
+        post_id = uuid.uuid4()
+        return PostDetail().put(request, author_id, post_id)
+
+class CommentList(ListCreateAPIView):
+    serializer_class = CommentSerializer
+    pagination_class = CommentsPagination
+
+    # used by the ListCreateAPIView super class 
+    def get_queryset(self):
+        return self.comments
+
+    """
+    get comments of the post
+    """
+    def get(self, request, *args, **kwargs):
+        try:
+            self.comments = Comment.objects.filter(
+                post_id=kwargs.get("post_id")
+            ).order_by('-published')
+        except (KeyError, Post.DoesNotExist):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+ 
+        response = super().list(request, *args, **kwargs)
+        return response
+
+    """
+    add comment to the post
+    """
+    def post(self, request, author_id, post_id):
+        try:
+            author = Author.objects.get(pk=author_id)
+            post = Post.objects.get(pk=post_id)
+        except (Author.DoesNotExist, Post.DoesNotExist):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        serializer = CommentSerializer(data=request.data)
+        if serializer.is_valid():
+            comment = Comment.objects.create(
+                author=author, 
```

Comment:
one way of doing this could be that we serialize and then save foreign author to database, but leave user as null. Do this to all the external objects, so we store them just as our own objects, the only difference being the Author object has no user.

Reply:
I think it is okay to serialize the author, as the format should follow the spec. The user = null should work, I'll test it out